### PR TITLE
LVNOD-83-gift-logic

### PR DESCRIPTION
**Pull request by denisbodnar**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-83)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-97)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-98)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-100)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-101)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-102)

## Code reviewers

- [ ] @yarvanya
- [ ] @shyyka
- [ ] @yuliannakorol

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

### CODE REVIEW

**Review by shyyka — DISMISSED** [review]

File: app/src/containers/AddGift/AddGift.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { Modal, Button, FormGroup, FormControl, ControlLabel,
+} from 'react-bootstrap';
```
Comment (yarvanya) [inline app/src/containers/AddGift/AddGift.js]: Delete unneeded comma.

File: app/src/containers/AddGift/AddGift.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+const FieldGroup = ({ id, label, ...props }) => (
+  <FormGroup controlId={id}>
+    <ControlLabel>{label}</ControlLabel>
+    <FormControl {...props} />
+  </FormGroup>
+);
+
+export default class AddGift extends React.Component {
+  constructor (props) {
```
Comment (yarvanya) [inline app/src/containers/AddGift/AddGift.js]: Delete space before (

File: app/src/containers/AddGift/AddGift.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    reader.onload = upload => {
+      self.setState({
+        image: upload.target.result
+      });
+    };
+    reader.readAsDataURL(file);
+  };
+
+  render () {
```
Comment (yarvanya) [inline app/src/containers/AddGift/AddGift.js]: Delete space before (

File: app/src/containers/EditGift/EditGift.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { Modal, Button, FormGroup, FormControl, ControlLabel,
```
Comment (yarvanya) [inline app/src/containers/EditGift/EditGift.js]: Delete unneeded comma.

File: app/src/containers/EditGift/EditGift.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    reader.onload = upload => {
+      self.setState({
+        image: upload.target.result
+      });
+    };
+    reader.readAsDataURL(file);
+  };
+
+  render () {
```
Comment (yarvanya) [inline app/src/containers/EditGift/EditGift.js]: Delete space before (

File: app/src/containers/EditGift/EditGift.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
```
Comment (yarvanya) [inline app/src/containers/EditGift/EditGift.js]: Delete this empty blank.

File: app/src/redux/gift.js [hunk 7]
```diff
@@ -0,0 +1,146 @@
+const FETCH_GIFTS_SUCCESS = 'FETCH_GIFTS_SUCCESS';
```
Comment (yarvanya) [inline app/src/redux/gift.js:3]: First should be importing and after that declareting constants.

File: app/src/redux/gift.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+const UPDATE_GIFT_SUCCESS = 'UPDATE_GIFT_SUCCESS';
+const UPDATE_GIFT_FAIL = 'UPDATE_GIFT_FAIL';
+
+import { API } from './../helper/constants';
+import request from './../helper/request';
+
+export const fetchGifts = eventId => {
+  return dispatch => {
+    return request()
+      .get(API.URL + `/api/events/${eventId}/gifts`)
```
Comment (yarvanya) [inline app/src/redux/gift.js]: `${API.URL}/api/events/${eventId}/gifts`

File: app/src/redux/gift.js [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+            payload: res.body
+          })
+      );
+  };
+};
+
+export const createGift = (eventId, gift) => {
+  return dispatch => {
+    return request()
+      .post(API.URL + `/api/events/${eventId}/gifts`)
```
Comment (yarvanya) [inline app/src/redux/gift.js]: The same.

File: app/src/redux/gift.js [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            payload: res.body
+          })
+      );
+  };
+};
+
+export const updateGift = (eventId, giftId, gift) => {
+  return dispatch => {
+    return request()
+      .put(API.URL + `/api/event/${eventId}/gift/${giftId}`)
```
Comment (yarvanya) [inline app/src/redux/gift.js]: The same.

File: app/src/redux/gift.js [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+            payload: res.body
+          })
+      );
+  };
+};
+
+export const deleteGift = (eventId, giftId) => {
+  return dispatch => {
+    return request()
+      .delete(API.URL + `/api/event/${eventId}/gift/${giftId}`)
```
Comment (yarvanya) [inline app/src/redux/gift.js]: The same.

File: app/src/redux/gift.js [hunk 12]
```diff
@@ -0,0 +1,147 @@
+import { API } from './../helper/constants';
+import request from './../helper/request';
```
Comment (4asof4ik) [inline app/src/redux/gift.js]: fix path

File: app/src/containers/Gift/Gift.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as giftActions from '../../redux/gift';
```
Comment (4asof4ik) [inline app/src/containers/Gift/Gift.js]: path

File: app/src/redux/gift.js [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+const DELETE_GIFT_FAIL = 'DELETE_GIFT_FAIL';
+const UPDATE_GIFT_SUCCESS = 'UPDATE_GIFT_SUCCESS';
+const UPDATE_GIFT_FAIL = 'UPDATE_GIFT_FAIL';
+
+export const fetchGifts = eventId => {
+  return dispatch => {
+    return request()
+      .get(`${API.URL}/api/events/${eventId}/gifts`)
+      .end((err, res) => err &&
+          dispatch({
```
Comment (4asof4ik) [inline app/src/redux/gift.js]: improve check all

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: app/src/containers/GiftList/GiftList.js [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+
+      const handleDelete = e => {
+        e.preventDefault();
+        this.props.actions.deleteGift(this.props.id, gift.id);
+      };
+
+      return (
+        <Panel header={gift.name} eventKey={gift.id} key={gift.id}>
+          {!!gift.image && <Image src={gift.image} responsive/>}
+          <p>Description:&nbsp;{gift.description}</p>
```
Comment (4asof4ik) [inline app/src/containers/GiftList/GiftList.js]: &nbsp ? you have scss

File: app/src/containers/GiftList/GiftList.js [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+          <p>Link:&nbsp;<a href={'http://' + gift.link}>{gift.link}</a></p>
```
Comment (4asof4ik) [inline app/src/containers/GiftList/GiftList.js]: `${}` and what if i need https

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by yarvanya — APPROVED** [review]

**Review by yuliannakorol — APPROVED** [review]

**Review by shyyka — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**